### DISCUSSION TITLE
sagemaker.processing.Processor run has no code input?

### DISCUSSION THREAD

**Discussion by entest-hai** [aws/sagemaker-python-sdk#3335] [Help]
What is the purpose of sagemaker.processing.Processor? since its run method does not accept any code or script? Does it mean Processor is just a base lass for others such as SklearnProcessor, FrameworProcessor?

**Comment by Sekhar-jami** [comment]
You can use it if you have your own custom processing code. You can create a docker image out of your custom code & pass it as a parameter:
`from sagemaker.processing import Processor


pre_processor = Processor(
    **image_uri=pre_processing_image_uri,**
    instance_type=pre_processing_instance_type,
    instance_count=pre_processing_instance_count,
    base_job_name=base_job_name,
    role=role,
    sagemaker_session=pipeline_session,
    tags=tags,
    network_config=network_config
)`

For more details refer: https://docs.aws.amazon.com/sagemaker/latest/dg/build-your-own-processing-container.html#byoc-run
